### DISCUSSION TITLE
does clickhouse-backup  support backup to HDFS？

### DISCUSSION THREAD

**Discussion by wangkeye852** [Altinity/clickhouse-backup#403] [Q&A] [answered]
does clickhouse-backup  support backup to HDFS ，or is there any trend of support

**Comment by Slach** [comment]
Do you mean backup volume which have type HDFS?
https://clickhouse.com/docs/en/operations/storing-data/#configuring-hdfs
No, currenlty it not implemented, and i'm not sure it will implements in near future, cause it hard to move data from hdfs to all supported remote storage types


Or you mean table with ENGINE=hdfs
https://clickhouse.com/docs/en/engines/table-engines/integrations/hdfs/
for table with `hdfs` engine, `clickhouse-backup` backup only SQL schema

Why do you need implements HDFS backup?

**Reply by haiya512** [reply]
@Slach  does it support hdfs right now ? backup volume which have type HDFS

**Reply by Slach** [reply, marked as answer]
@haiya512 
currently, you could try to use

```yaml
clickhouse:
  use_embedded_backup_restore: true
```

for backup hdfs volumes

but embedded backup is not support incremental backup right now
